### PR TITLE
test: fix tests for #251

### PR DESCRIPTION
**Pull request by xukai92**
addressing #254

i left comments in the code diff

### CODE REVIEW

File: test/frontend/cast.jl [hunk 1]
```diff
@@ -38,6 +38,7 @@ options = [
     JLOption(:option1, false, Any, "nothing"),
     JLOption(:option2, false, Int, "1::Int"),
     JLOption(:option3, false, String, "abc::String"),
+    JLOption(:option4, false, Int, "1::Int"),
```
Comment (xukai92) [inline test/frontend/cast.jl:41]: i missed this in my previous change

File: test/frontend/cast.jl [hunk 2]
```diff
@@ -103,7 +104,7 @@ function foo(
     @test cmd.options["option4"].hint == "1::Int"
 
     @show cmd.options
-    @test all(keys(cmd.options) .== ["option1", "o", "option2", "option3"])
+    @test all(keys(cmd.options) .== ["option1", "o", "option2", "option3", "option4"])
```
Comment (xukai92) [inline test/frontend/cast.jl:107]: corresponding update

File: test/frontend/cast.jl [hunk 3]
```diff
@@ -119,6 +120,7 @@ end
             option1 = nothing,
             option2::Int = 1,
             option3::String = "abc",
+            option4::Int = 1,
```
Comment (xukai92) [inline test/frontend/cast.jl:123]: corresponding update

File: test/frontend/cast.jl [hunk 4]
```diff
@@ -91,7 +92,7 @@ function foo(
     @test cmd.options["option1"].name == "option1"
     @test cmd.options["option2"].name == "option2"
     @test cmd.options["o"].short == true
-    @test cmd.options["o"].hint === nothing
+    @test cmd.options["o"].hint === "nothing"
```
Comment (xukai92) [inline test/frontend/cast.jl:95]: i'm not 100% sure about this. hint is suppose to be a `String` so I think the previous test was wrong?

Comment (Roger-luo) [reply test/frontend/cast.jl:95]: yes, previously the hint will be missing so it wasn't string but nothing... but I think now this is the right fix

**Comment by Roger-luo** [comment]
Thanks for the quick fix!

**Pull request merged**